aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -180,26 +182,32 @@ NestedTensorImpl::NestedTensorImpl(
       "in the near future.");
   auto storage_device = storage_.device();
   TORCH_INTERNAL_ASSERT(
-      storage_device.is_cpu() || storage_device.is_cuda(),
-      "NestedTensorImpl storage must be either CUDA or CPU but got ",
+      storage_device.is_cpu() || storage_device.is_cuda() || storage_device.is_meta(),
+      "NestedTensorImpl storage must be CUDA, CPU, or Meta; got ",
```

Comment:
I don't really understand why this device check is needed at all. Why not just work polymorphically over all devices.

Reply:
Leftover historical artifact, I don't see a need for it either, removing.

In general, a lot of NT code seems to run very tight checks ensuring we don't leave the narrow region we expected to support